prometheus-export/src/main/scala/StreamingQuerySource.scala

```diff
@@ -202,6 +264,33 @@ class StreamingQuerySource() extends Source with Logging {
             }
           }
         }
+
+        case "IcebergMicroBatchStream" => {
+          reportHistogram(
+            "streaming.source.iceberg.rows",
+            commonLabels + ("source" -> name),
+            source.numInputRows
+          )
+
+          val latestOffset = icebergOffset(source.endOffset)
+          if (latestOffset.nonEmpty) {
+            reportGauge(
+              s"streaming.source.icberg.partition.offset",
+              commonLabels + ("offset_type" -> "latestOffset"),
+              latestOffset.get("snapshot_id").get.left.get
```

Comment:
`Map` has an `apply` if you know the entry is gonna be there.

```suggestion
              latestOffset("snapshot_id").left.get
```